packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -138,9 +171,30 @@ class _BottomAppBarState extends State<BottomAppBar> {
           notchMargin: widget.notchMargin,
         )
       : const ShapeBorderClipper(shape: RoundedRectangleBorder());
-    final double elevation = widget.elevation ?? babTheme.elevation ?? _defaultElevation;
-    final Color color = widget.color ?? babTheme.color ?? Theme.of(context).bottomAppBarColor;
+    final double elevation = widget.elevation ?? babTheme.elevation ?? defaults.elevation ?? _defaultElevation;
+    final double? height = widget.height ?? babTheme.height ?? defaults.height;
+    final Color color = widget.color ?? babTheme.color ?? defaults.color ?? theme.bottomAppBarColor;
+    final Color surfaceTintColor = widget.surfaceTintColor ?? babTheme.surfaceTintColor
+        ?? defaults.surfaceTintColor ?? theme.colorScheme.surfaceTint;
     final Color effectiveColor = ElevationOverlay.applyOverlay(context, color, elevation);
+
+    if (isMaterial3) {
+      return SizedBox(
+        height: height,
+        child: Material(
+          key: materialKey,
+          color: color,
+          elevation: elevation,
+          surfaceTintColor: surfaceTintColor,
+          clipBehavior: widget.clipBehavior,
+          child: SafeArea(child: Padding(
```

Comment:
The convention in this repo seems to be one Widget per line, so I'd put the Padding widget on the next line.